### PR TITLE
etcdctlv3: add formatting for json and protobuf

### PR DESCRIPTION
**Pull request by heyitsanthony**

### CODE REVIEW

File: etcdctlv3/README.md [hunk 1]
```diff
@@ -13,13 +13,19 @@ PUT assigns the specified value with the specified key. If key already holds a v
...
+##### Simple reply
 
 - OK if PUT executed correctly. Exit code is zero.
 
 - Error string if PUT failed. Exit code is non-zero.
 
-TODO: probably json and binary encoded proto
+##### JSON reply
+
+The JSON encoding of the PUT RPC response.
```
Comment (xiang90) [inline etcdctlv3/README.md]: Link to the PRC response definition?

File: etcdctlv3/README.md [hunk 2]
```diff
@@ -13,13 +13,19 @@ PUT assigns the specified value with the specified key. If key already holds a v
...
+
+##### Protobuf reply
+
+The protobuf encoding of the PUT RPC response.
```
Comment (xiang90) [inline etcdctlv3/README.md]: Link to the PRC response definition?

File: etcdctlv3/command/printer.go [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+	for _, kv := range r.Kvs {
+		printJSON(kv)
+	}
+}
+func (p *jsonPrinter) Put(r v3.PutResponse)     { printJSON(r) }
+func (p *jsonPrinter) Txn(r v3.TxnResponse)     { printJSON(r) }
+func (p *jsonPrinter) Watch(r v3.WatchResponse) { printJSON(r) }
+
+func printJSON(v interface{}) {
+	b, err := json.Marshal(v)
```
Comment (xiang90) [inline etcdctlv3/command/printer.go]: do we need to convert the key.value in json format to hex if hex is specified since json is also text based.

Comment (heyitsanthony) [inline etcdctlv3/command/printer.go]: Based on my tests, it looks like it's base64 encoding it.

File: etcdctlv3/README.md (hunk 2 above)
Comment (heyitsanthony) [inline etcdctlv3/README.md]: sure

**Comment by heyitsanthony** [comment]
fixed up ptal /cc @xiang90

File: etcdctlv3/README.md [hunk 4]
```diff
@@ -3,6 +3,9 @@ etcdctl
 
 ## Commands
 
+[etcdrpc]: ../etcdserver/etcdserverpb/rpc.proto
```
Comment (xiang90) [inline etcdctlv3/README.md]: move this to the bottom of the doc? we usually put this at the bottom. :)

**Comment by xiang90** [comment]
LGTM

File: etcdctlv3/README.md (hunk 4 above)
Comment (heyitsanthony) [inline etcdctlv3/README.md]: sure

**Pull request merged**